### PR TITLE
add mongo config to server

### LINKED ISSUE #15: Have mongo connection settings be configurable
```
MONGO_HOST=
MONGO_USER=
MONGO_PASS=
MONGO_DB=
```

### PR DESCRIPTION
**Pull request by cheapsteak**
closes #15

### CODE REVIEW

**Comment by Jephuff** [comment]
hmm, i'm getting a message about user and pass not being supported.
image

**Comment by cheapsteak** [comment]
Interesting
These guys seem to have a similar issue
https://github.com/Automattic/mongoose/issues/5419

Might need to change to credentials in the URI

**Comment by cheapsteak** [comment]
updated @Jephuff

File: index.ts [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-
 import { start } from './server';
 
-dotenv.config();
-
 mongoose.Promise = global.Promise;
-mongoose.connect('mongodb://localhost/test', { useMongoClient: true });
+
+const { mongoUser, mongoPass } = config;
+const mongoUri = `mongodb://${encodeURIComponent(
```
Comment (Jephuff) [inline index.ts]: Thoughts on making user and password optional?

Comment (cheapsteak) [reply index.ts]: done

**Pull request merged**